webapp/portlet/src/main/webapp/vue-apps/version-history-drawer/components/VersionCard.vue

```diff
@@ -0,0 +1,223 @@
+              class="px-1"
+              small
+              @click="updateDescription">
+              fa-check
+            </v-icon>
+            <v-icon
+              class="clickable px-0"
+              color="red"
+              small
+              @click="resetInput">
+              fa-times
+            </v-icon>
+          </div>
+        </v-text-field>
+      </div>
+      <div
+        v-show="descriptionInputHidden"
+        v-if="!versionObject.current && canManage && !disableRestoreVersion"
+        class="item-version-restore">
+        <v-progress-circular
+          v-if="isRestoringVersion"
+          :size="20"
+          color="primary"
+          indeterminate />
+        <v-tooltip
+          v-if="!isRestoringVersion"
+          bottom>
+          <template #activator="{ on, attrs }">
+            <v-icon
+              v-bind="attrs"
+              v-on="on"
+              size="22"
+              class="primary--text clickable pa-0 mt-1"
+              @click="restoreVersion">
+              mdi-restart
+            </v-icon>
+          </template>
+          <span class="caption">{{ $t('versionHistory.label.restore') }}</span>
+        </v-tooltip>
+      </div>
+    </div>
+  </v-main>
+</template>
+
+<script>
+export default {
+  props: {
+    version: {
+      type: Object,
+      default: () => {
+        return {};
+      }
+    },
+    enableEditDescription: {
+      type: Boolean,
+      default: () => {
+        return false;
+      },
+    },
+    disableRestoreVersion: {
+      type: Boolean,
+      default: () => {
+        return false;
+      },
+    },
+    canManage: {
+      type: Boolean,
+      default: () => {
+        return false;
+      }
+    }
+  },
+  data: () => ({
+    dateTimeFormat: {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+      hour: '2-digit',
+      minute: '2-digit'
+    },
+    MAX_DESCRIPTION_LENGTH: 500,
+    descriptionInputHidden: true,
+    newDescription: '',
+    versionObject: {},
+    isUpdatingDescription: false,
+    isRestoringVersion: false
+  }),
+  created() {
+    this.$root.$on('version-restored', (restoredVersion) => {
+      this.versionObject.current = this.versionObject.id === restoredVersion.id;
+      this.isRestoringVersion = false;
+    });
+    this.$root.$on('version-restore-error', () => {
+      this.isRestoringVersion = false;
+    });
+    this.$root.$on('version-description-updated', (version) => {
+      if (version.id === this.version.id) {
+        this.versionObject.summary = version.summary;
+        this.isUpdatingDescription = false;
+        this.descriptionInputHidden = true;
+      }
+    });
+    this.$root.$on('version-description-update-error', (version) => {
+      if (version.id === this.version.id) {
+        this.isUpdatingDescription = false;
+        this.descriptionInputHidden = false;
+      }
+    });
+    this.versionObject = Object.assign({}, this.version);
+  },
```

Comment:
```suggestion
  created() {
    this.$root.$on('version-restored', this.handleVersionRestore);
    this.$root.$on('version-restore-error', this.handleVersionRestoreError);
    this.$root.$on('version-description-updated', this.handleDescriptionUpdated);
    this.$root.$on('version-description-update-error', this.handleDescriptionUpdateError);
    this.versionObject = Object.assign({}, this.version);
  },
  beforeDestroy: {
    this.$root.$off('version-restored', this.handleVersionRestore);
    this.$root.$off('version-restore-error', this.handleVersionRestoreError);
    this.$root.$off('version-description-updated', this.handleDescriptionUpdated);
    this.$root.$off('version-description-update-error', this.handleDescriptionUpdateError);
  },
```

Use functions for events management on reusable components especially that it's destroyed when switching when browsing from a note to another. This way, you can delete listener in `beforeDestroy`. else, the JS Garbage Collector will not be able to delete those listeners and components from memory that will increase considerably when using this UI.